### PR TITLE
Add `error-stack-macros` crate

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We may want to support proc-macros for error stack. In order to reserve the name, we publish a crate for this.

## 🔍 What does this change?

- Add a `error-stack-macros` crate
- Adjust CI to support publishing of sub-crates
- Make `error-stack`s _Makefile.toml_ compatible with `--workspace`
- Drive-by: Fix MSRV in `Cargo.toml`

## ⚠️ Known issues

Our CI requires a `rust-toolchain.toml` file in the directory. This could be solved by investigating the output of `cargo metadata` but this is out of scope for this PR. This may be useful later on but is currently not on the roadmap. For now, I added a comment in `error-stack/rust-toolchain.toml` file to also update the macros.

### CODE REVIEW

File: packages/libs/error-stack/Makefile.toml [hunk 1]
```diff
@@ -1,27 +1,78 @@
 extend = { path = "../../../.github/scripts/rust/Makefile.toml" }
```
Comment (TimDiekmann) [inline packages/libs/error-stack/Makefile.toml:1]: I expect changes to this file to be reverted when `DebugDiagnostics` and the `hooks` feature is removed from `error-stack`

File: packages/libs/error-stack/Cargo.toml [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+members = [".", "macros"]
+default-members = ["."]
+
 [package]
 name = "error-stack"
 version = "0.1.1"
 authors = ["HASH"]
 edition = "2021"
-rust-version = "1.61.0"
+rust-version = "1.63.0"
```
Comment (Alfred-Mountfield) [inline packages/libs/error-stack/Cargo.toml:10]: Do you mean to implicitly update this as part of this PR?

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -208,7 +208,7 @@ jobs:
 
       - name: Run lints
         working-directory: ${{ matrix.directory }}
-        run: cargo +${{ matrix.toolchain }} make --profile ${{ matrix.profile }} clippy -- -D warnings
+        run: cargo +${{ matrix.toolchain }} make --profile ${{ matrix.profile }} lint
```
Comment (Alfred-Mountfield) [inline .github/workflows/rust.yml:211]: Is this equivalent?

**Review by Alfred-Mountfield — COMMENTED** [review]
Generally fine with these, considering we're about to address the state of the Makefile

File: packages/libs/error-stack/Cargo.toml (hunk 2 above)
Comment (TimDiekmann) [reply packages/libs/error-stack/Cargo.toml:10]: This was already done in https://github.com/hashintel/hash/pull/944 and https://github.com/hashintel/hash/pull/969, I just forgot that this is also captured in `Cargo.toml`.

File: .github/workflows/rust.yml (hunk 3 above)
Comment (TimDiekmann) [reply .github/workflows/rust.yml:211]: Lint will do `rustfmt`, `clippy` and `doc`, so it's a superset. `cargo make doc` is not tested (and failed), only `cargo make rustdoc` is tested.

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**